xdsl/dialects/mod_arith.py

```diff
@@ -0,0 +1,74 @@
+"""
+mod_arith is a dialect implementing modular arithmetic, originally
+implemented as part of the HEIR project (https://github.com/google/heir/tree/main).
+"""
+
+from abc import ABC
+from typing import Annotated, Generic, TypeVar
+
+from xdsl.dialects.arith import signlessIntegerLike
+from xdsl.dialects.builtin import AnyIntegerAttr
+from xdsl.ir import Attribute, Dialect, Operation, SSAValue
+from xdsl.irdl import (
+    ConstraintVar,
+    IRDLOperation,
+    ParsePropInAttrDict,
+    irdl_op_definition,
+    operand_def,
+    prop_def,
+    result_def,
+)
+from xdsl.traits import Pure
+
+
+class ModArithOp(IRDLOperation, ABC):
+    pass
```

Comment:
I'm not sure that this is necessary, I'd advise against abstracting things before there's a real use for it.